controllers/nodefeaturediscovery_controls.go

```diff
@@ -306,6 +307,22 @@ func DaemonSet(n NFD) (ResourceStatus, error) {
 		obj.Spec.Template.Spec.Containers[0].ImagePullPolicy = n.ins.Spec.Operand.ImagePolicy(n.ins.Spec.Operand.ImagePullPolicy)
 	}
 
+	// check if running as instance
+	// https://kubernetes-sigs.github.io/node-feature-discovery/v0.8/advanced/master-commandline-reference.html#-instance
+	if n.ins.Spec.Instance != "" && obj.ObjectMeta.Name == "nfd-master" {
+		instanceFlag := fmt.Sprintf("--instance=%s", n.ins.Spec.Instance)
+		var found bool
+		for _, c := range obj.Spec.Template.Spec.Containers[0].Command {
+			if c == instanceFlag {
+				// all set!
+				found = true
+			}
+		}
```

Comment:
stupid question: can the instance flag change so that this code is unable to find it?

Reply:
Good catch!

Reply:
Hmm, could we drop this fragile-looking logic altogether and just forcefully (re-)set the container command line?

E.g. so that `build/assets/master/0400_master_daemonset.yaml` would have simply `command: ["nfd-master"]` and we would set `obj.Spec.Template.Spec.Containers[0].Args` to `{"--port=12000"}` here and append `--instance` if necessary. Would make it a lot cleaner and more robust IMO. WDYT?

Reply:
agree, also TODO allow users to define the master port via CRD, WDYT? 
cc @zvonkok

Reply:
Yes of course. I'm not sure how useful that would be 🤔 Maybe it is